flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveCatalogUtil.java

```diff
@@ -0,0 +1,235 @@
+	 */
+	public static ColumnStatistics createTableColumnStats(
+			Table hiveTable,
+			Map<String, CatalogColumnStatisticsDataBase> colStats) {
+		ColumnStatisticsDesc desc = new ColumnStatisticsDesc(true, hiveTable.getDbName(), hiveTable.getTableName());
+		return createColumnStatistics(colStats, hiveTable.getSd(), desc);
+	}
+
+	/**
+	 * Create columnStatistics from the given Hive column stats of a hive partition.
+	 */
+	public static ColumnStatistics createPartitionColumnStats(
+			Partition hivePartition,
+			String partName,
+			Map<String, CatalogColumnStatisticsDataBase> colStats) {
+		ColumnStatisticsDesc desc = new ColumnStatisticsDesc(false, hivePartition.getDbName(), hivePartition.getTableName());
+		desc.setPartName(partName);
+		return createColumnStatistics(colStats, hivePartition.getSd(), desc);
+	}
+
+	private static ColumnStatistics createColumnStatistics(
+			Map<String, CatalogColumnStatisticsDataBase> colStats,
+			StorageDescriptor sd,
+			ColumnStatisticsDesc desc) {
+		List<ColumnStatisticsObj> colStatsList = new ArrayList<>();
+
+		for (FieldSchema field : sd.getCols()) {
+			String hiveColName = field.getName();
+			String hiveColType = field.getType();
+
+			if (colStats.containsKey(hiveColName)) {
+				CatalogColumnStatisticsDataBase flinkColStat = colStats.get(field.getName());
+				ColumnStatisticsData statsData =
+						getColumnStatisticsData(HiveTypeUtil.toFlinkType(TypeInfoUtils.getTypeInfoFromTypeString(hiveColType)), flinkColStat);
+				ColumnStatisticsObj columnStatisticsObj = new ColumnStatisticsObj(hiveColName, hiveColType, statsData);
+				colStatsList.add(columnStatisticsObj);
+			}
+		}
+
+		return new ColumnStatistics(desc, colStatsList);
+	}
+
+	/**
+	 * Create Flink ColumnStats from Hive ColumnStatisticsData.
+	 */
+	private static CatalogColumnStatisticsDataBase createTableColumnStats(DataType colType, ColumnStatisticsData stats) {
+		if (stats.isSetBinaryStats()) {
+			BinaryColumnStatsData binaryStats = stats.getBinaryStats();
+			return new CatalogColumnStatisticsDataBinary(
+					binaryStats.getMaxColLen(),
+					binaryStats.getAvgColLen(),
+					binaryStats.getNumNulls());
+		} else if (stats.isSetBooleanStats()) {
+			BooleanColumnStatsData booleanStats = stats.getBooleanStats();
+			return new CatalogColumnStatisticsDataBoolean(
+					booleanStats.getNumFalses(),
+					booleanStats.getNumFalses(),
+					booleanStats.getNumNulls());
+		} else if (stats.isSetDateStats()) {
+			DateColumnStatsData dateStats = stats.getDateStats();
+			return new CatalogColumnStatisticsDataDate(
+					new org.apache.flink.table.catalog.stats.Date(dateStats.getLowValue().getDaysSinceEpoch()),
+					new org.apache.flink.table.catalog.stats.Date(dateStats.getHighValue().getDaysSinceEpoch()),
+					dateStats.getNumDVs(),
+					dateStats.getNumNulls());
+		} else if (stats.isSetDoubleStats()) {
+				DoubleColumnStatsData doubleStats = stats.getDoubleStats();
+				return new CatalogColumnStatisticsDataDouble(
```

Comment:
minor: to improve readability, how about reformat it to 

```
return new CatalogColumnStatisticsDataDouble(
						doubleStats.getLowValue(),
						doubleStats.getHighValue(),
						doubleStats.getNumDVs(),
						doubleStats.getNumNulls());
```

Same to code below